packages/web-components/src/menu/README.md

```diff
@@ -0,0 +1,99 @@
+# Menu
+
+A popup or contextual menu displays a list of options on a temporary surface. They are invoked when users interact with a button, action, or other control.
+
+## Design Spec
+
+[Link to Menu Design Spec in Figma](https://www.figma.com/file/jFWrkFq61GDdOhPlsz6AtX/Menu?node-id=1528%3A5102&t=XtW4laeEzgVFIl1E-0)
+
+## Engineering Spec
+
+Fluent WC3 Menu extends from the FAST Menu [FAST Menu](https://explore.fast.design/components/fast-menu) and is intended to be as close to the Fluent UI React 9 Menu implementation as possible. However, due to the nature of web components there will not be 100% parity between the two.
```

Comment:
I think we'll need to investigate here. Menu trigger is called out as "Menu Button" in the ARIA Practices Guidance. The semantic _menu_ itself is the element with a role of menu. Before we decide that 100% alignment is the best approach here, I think we should look at how this might compose first and decide which abstractions are helpful here.

Reply:
@chrisdholt We are planning on the development of a `PopOver` component, correct? Suggested composure:
```ts
<button id="menu-button" role="button" aria-haspopup="menu" aria-expanded="false" aria-controls="menu-popover">Menu</button>

<popover id="menu-popover" aria-orientation="vertical" aria-labelledby="menu-button" aria-hidden="true">
  <menu role="menu">
        <menuitem role="menuitem">Option 1</menuitem>
        <menuitem role="menuitem">Option 2</menuitem>
  </menu>
</popover>
```

Reply:
Currently the menu has support with Floating UI without that abstraction given that `popover` from a platform standpoint is not intended to be an HTML element but rather an attribute which can be combined with CSS Anchored Positioning. Can we grab the react spec and find the history of the PR to dig into some of the decisions here? Again, I think we can align with "MenuList" if we need to, but the role on the element is *menu* and that is the breakdown the APG uses rather than MenuButton + MenuList, Menu is the thing, the broader composition includes multiple roles.

Reply:
There are ways we can break down the FAST implementation of Menu to support the variations above (for instance if we need to have an inline menu which doesn't "float", but my concern is that adding separate complete component abstractions here create performance considerations (more elements for one)...as with others I would prefer that we understand the reasoning here and translate that to something which makes sense for WC's. Elements come at a cost, including creation. How can we support this without taking a perf hit to match a react construct. To be clear here, the delta is that the "components" above are JSX, not unique elements within the browser needing to be registered, etc. While we could create component abstractions for each of these, I don't think that follows in the spirit of how we'd actually build this as a WC given encapsulation, etc. So let's try to understand the intent here and identify "parity" given earned differences. Otherwise, we're taking on cost to match an implementation detail of a framework construct (React's component model vs. Web Components).

Reply:
Agree that JSX is something different than DOM. Let's do what works in Web Components and then compare (and eventually refactor)

Reply:
@ling1726 @chrisdholt  @miroslavstastny Any ideas where I might find the react menu spec?

Reply:
@brianchristopherbrady  https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-menu/docs/Spec.md